### PR TITLE
JMX enable. Enable KAFKA_JMX_OPTS and JMX_PORT in marathon.json.mustache

### PR DESCRIPTION
**Pull request by mbdcos**

### CODE REVIEW

**Comment by mbdcos** [comment]
retest this please

File: frameworks/kafka/universe/marathon.json.mustache [hunk 1]
```diff
@@ -42,13 +42,10 @@
...
-    "JMX_ENABLE": "{{brokers.jmx.enable}}",
-    "JMX_REMOTE_ENABLE": "{{brokers.jmx.remote}}",
-    "JMX_REMOTE_PORT": "{{brokers.jmx.remote_port}}",
-    "JMX_REMOTE_REGISTRY_SSL": "{{brokers.jmx.remote_registry_ssl}}",
-    "JMX_REMOTE_SSL": "{{brokers.jmx.remote_ssl}}",
-    "JMX_REMOTE_AUTH": "{{brokers.jmx.remote_authenticate}}",
-    "JMX_REMOTE_SSL_NEED_CLIENT_AUTH": "{{brokers.jmx.remote_ssl_need_client_auth}}",
+    {{#brokers.jmx.enable}}
+     "TASKCFG_ALL_KAKFA_JMX_OPTS": "-Dcom.sun.management.jmxremote={{brokers.jmx.remote}} -Dcom.sun.management.jmxremote.registry.ssl={{brokers.jmx.remote_registry_ssl}} -Dcom.sun.management.jmxremote.ssl={{brokers.jmx.remote_ssl}} -Dcom.sun.management.jmxremote.ssl.need.client.auth={{brokers.jmx.remote_ssl_need_client_auth}} -Dcom.sun.management.jmxremote.authenticate={{brokers.jmx.remote_authenticate}} {{brokers.additional_kafka_jmx_opts}}"
+     "TASKCFG_ALL_JMX_PORT": "{{brokers.jmx.remote_port}}",
```
Comment (gabrielhartmann) [inline frameworks/kafka/universe/marathon.json.mustache]: The port isn't being reserved: https://github.com/mesosphere/dcos-commons/blob/master/frameworks/kafka/src/main/dist/svc.yml#L17-L19

Comment (mbdcos) [inline frameworks/kafka/universe/marathon.json.mustache]: what is your opinion about reserving JMX_PORT. JMX_PORT is defined only if jmx.enable is true.  svc.yml.mustache is an option.

Comment (mbdcos) [inline frameworks/kafka/universe/marathon.json.mustache]: JMX_PORT is in kafak-start.sh, we can not change the name.
"env-key" in ports should work to do this in svc.yml

**Comment by mrbrowning** [comment]
@mbdcos could you break out the env key changes into its own PR?

File: frameworks/kafka/universe/config.json [hunk 2]
```diff
@@ -142,7 +142,12 @@
                 "description": "Whether to enable SSL client authentication for remote access",
                 "type": "boolean",
                 "default": false
-              }
+              },
+              "additional_kafka_jmx_opts": {
```
Comment (gabrielhartmann) [inline frameworks/kafka/universe/config.json]: I like this.

File: frameworks/kafka/universe/marathon.json.mustache [hunk 3]
```diff
@@ -42,13 +42,10 @@
...
     "MESOS_ZOOKEEPER_URI": " master.mesos:2181",
-    "JMX_ENABLE": "{{brokers.jmx.enable}}",
-    "JMX_REMOTE_ENABLE": "{{brokers.jmx.remote}}",
-    "JMX_REMOTE_PORT": "{{brokers.jmx.remote_port}}",
-    "JMX_REMOTE_REGISTRY_SSL": "{{brokers.jmx.remote_registry_ssl}}",
-    "JMX_REMOTE_SSL": "{{brokers.jmx.remote_ssl}}",
-    "JMX_REMOTE_AUTH": "{{brokers.jmx.remote_authenticate}}",
-    "JMX_REMOTE_SSL_NEED_CLIENT_AUTH": "{{brokers.jmx.remote_ssl_need_client_auth}}",
+    {{#brokers.jmx.enable}}
+     "TASKCFG_ALL_KAKFA_JMX_OPTS": "-Dcom.sun.management.jmxremote={{brokers.jmx.remote}} -Dcom.sun.management.jmxremote.registry.ssl={{brokers.jmx.remote_registry_ssl}} -Dcom.sun.management.jmxremote.ssl={{brokers.jmx.remote_ssl}} -Dcom.sun.management.jmxremote.ssl.need.client.auth={{brokers.jmx.remote_ssl_need_client_auth}} -Dcom.sun.management.jmxremote.authenticate={{brokers.jmx.remote_authenticate}} {{brokers.additional_kafka_jmx_opts}}"
```
Comment (gabrielhartmann) [inline frameworks/kafka/universe/marathon.json.mustache]: You can do more nicely formatted multi-line strings in YAML: http://stackoverflow.com/a/3790497

**Comment by mbdcos** [comment]
retest this please

**Pull request merged**